sdk/python/feast/infra/materialization/kubernetes/k8s_materialization_engine.py

```diff
@@ -0,0 +1,421 @@
+                    self.v1.read_namespaced_pod_log(pod.metadata.name, self.namespace)
+                )
+            except ApiException as e:
+                logger.warning(f"Could not retrieve pod logs due to: {e.body}")
+
+    def _create_kubernetes_job(self, job_id, paths, feature_view):
+        try:
+            # Create a k8s configmap with information needed by pods
+            self._create_configuration_map(job_id, paths, feature_view, self.namespace)
+
+            # Create the k8s job definition
+            self._create_job_definition(
+                job_id=job_id,
+                namespace=self.namespace,
+                pods=len(paths),  # Create a pod for each parquet file
+                env=self.batch_engine_config.env,
+            )
+            job = KubernetesMaterializationJob(job_id, self.namespace)
+            logger.info(f"Created job `{job.job_id()}` on namespace `{self.namespace}`")
+            return job
+        except FailToCreateError as failures:
+            return KubernetesMaterializationJob(job_id, self.namespace, error=failures)
+
+    def _create_configuration_map(self, job_id, paths, feature_view, namespace):
+        """Create a Kubernetes configmap for this job"""
+
+        feature_store_configuration = yaml.dump(self.repo_config.dict(by_alias=True))
+
+        materialization_config = yaml.dump(
+            {"paths": paths, "feature_view": feature_view.name}
+        )
+
+        labels = {"feast-materializer": "configmap"}
+        configmap_manifest = {
+            "kind": "ConfigMap",
+            "apiVersion": "v1",
+            "metadata": {
+                "name": self._configmap_name(job_id),
+                "labels": {**labels, **self.batch_engine_config.labels},
+            },
+            "data": {
+                "feature_store.yaml": feature_store_configuration,
+                "materialization_config.yaml": materialization_config,
+            },
+        }
+        self.v1.create_namespaced_config_map(
+            namespace=namespace,
+            body=configmap_manifest,
+        )
+
+    def _configmap_name(self, job_id):
+        return f"feast-{job_id}"
+
+    def _create_job_definition(self, job_id, namespace, pods, env, index_offset=0):
+        """Create a kubernetes job definition."""
+        job_env = [
+            {
+                "name": "MINI_BATCH_SIZE",
+                "value": str(self.batch_engine_config.mini_batch_size),
+            },
+        ]
+        # Add any Feast configured environment variables
+        job_env.extend(env)
+
+        securityContextCapabilities = None
+        if self.batch_engine_config.include_security_context_capabilities:
+            securityContextCapabilities = {
+                "add": ["NET_BIND_SERVICE"],
+                "drop": ["ALL"],
+            }
+
+        job_labels = {"feast-materializer": "job"}
+        pod_labels = {"feast-materializer": "pod"}
+        job_definition = {
```

Comment:
Can take this out as a template file, also make it a config option for the user to override the template